apps/teams-test-app/src/public/externalOauth_end.html

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <h2 style="font-size: 40px">Starting mock external oauth2</h2>
+  </head>
+  <body>
+    <script>
+      const failureMessage = {
+        type: 'authenticationFailed',
+        data: 'error',
+      };
+      const successMessage = {
+        type: 'authenticationSuccess',
+        data: 'success',
+      };
+
+      const params = new URLSearchParams(window.location.search);
+      const titleId = params.get('titleId');
+
+      if (window.opener) {
+        if (titleId === 'testOauthFailureTitleId') {
+          window.opener.postMessage(failureMessage, '*');
+        } else {
+          window.opener.postMessage(successMessage, '*');
+        }
+        setTimeout(() => {
+          window.close();
+        }, 2000);
```

Comment:
Can you please remove and let host SDK close the window as part of the validation?